services/ark-api/ark-api/src/ark_api/api/v1/openai.py

```
@@ -87,22 +87,50 @@ def process_request_metadata(
                     "error": {
                         "message": f"Invalid Ark metadata: {str(e)}",
                         "type": "invalid_request_error",
-                        "code": "invalid_ark_metadata"
+                        "code": "invalid_ark_metadata",
                     }
-                }
+                },
             )
     # Ignore other metadata keys per OpenAI SDK pattern
     return None
 
 
+# TODO: Maybe we should start streaming first, wait for the first chunk/response
+# and use the status code of that to respond with
 async def proxy_streaming_response(streaming_url: str):
     """Proxy streaming chunks from memory service."""
     timeout = httpx.Timeout(10.0, read=None)  # 10s connect, infinite read
     async with httpx.AsyncClient(timeout=timeout) as client:
         async with client.stream("GET", streaming_url) as response:
             if response.status_code != 200:
+                # Try to extract error message from response body
```

given we control the error structure, let's jsut read it with the expected structure and then fail otherwise, rather than trying fallbacks etc